assemble fails in task jsonp.1.0:jar: Files do not exist

Attempting to follow the contributing instructions on macOS 10.12. Cloning and `cnf:initialize` work fine, but `assemble` or `releaseNeeded` fail with `Project com.ibm.websphere.javaee.jsonp.1.0 has errors, 24 errors were reported`. I note that the files all have names starting with "ibm" and that it is looking at the JRE on my `$PATH` which is an Oracle JRE.

Full output:

```
PowerBook-i73000:dev liamwestby$ ./gradlew --stacktrace assemble

> Configure project :com.ibm.websphere.javaee.management.j2ee.1.1
-d /Users/liamwestby/Documents/Programming/open-liberty/dev/com.ibm.websphere.javaee.management.j2ee.1.1/build/classes/java/main
-cp /Users/liamwestby/Documents/Programming/open-liberty/dev/com.ibm.websphere.javaee.management.j2ee.1.1/build/classes/java/main:/Users/liamwestby/Documents/Programming/open-liberty/dev/com.ibm.websphere.javaee.ejb.3.2/build/libs/com.ibm.websphere.javaee.ejb.3.2.jar:/Users/liamwestby/Documents/Programming/open-liberty/dev/com.ibm.ws.org.osgi.annotation.versioning/build/libs/com.ibm.ws.org.osgi.annotation.versioning.jar

> Task :com.ibm.websphere.javaee.jsonp.1.0:jar
Warning: There is an empty jar or directory on the classpath: main
Warning: Imports that lack version ranges: [org.glassfish.json]
Warning: Unreferenced class path entries [javax.json-1.0.4, main]
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmcertpathfw.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjcefw.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjgssfw.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjgssprovider.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjssefw.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmorbapi.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmpkcs.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/math.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/rt.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/bin/default/jclSC180/vm.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/xml.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/xmldsigfw.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmcertpathfw.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjcefw.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjgssfw.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjgssprovider.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjssefw.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmorbapi.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmpkcs.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/math.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/rt.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/bin/default/jclSC180/vm.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/xml.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/xmldsigfw.jar


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':com.ibm.websphere.javaee.jsonp.1.0:jar'.
> Project com.ibm.websphere.javaee.jsonp.1.0 has errors, 24 errors were reported

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':com.ibm.websphere.javaee.jsonp.1.0:jar'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:242)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:95)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:235)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:224)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:121)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:77)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:102)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:96)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:612)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:567)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:96)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.api.GradleException: Project com.ibm.websphere.javaee.jsonp.1.0 has errors, 24 errors were reported
        at aQute.bnd.gradle.BndPlugin.checkErrors(BndPlugin.groovy:515)
        at aQute.bnd.gradle.BndPlugin.checkErrors(BndPlugin.groovy)
        at aQute.bnd.gradle.BndPlugin.this$2$checkErrors(BndPlugin.groovy)
        at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:479)
        at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:191)
        at org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:80)
        at org.gradle.internal.metaobject.BeanDynamicObject$GroovyObjectAdapter.invokeOpaqueMethod(BeanDynamicObject.java:579)
        at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:506)
        at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:191)
        at org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:80)
        at aQute.bnd.gradle.BndPlugin$_apply_closure1_closure13_closure48.doCall(BndPlugin.groovy:264)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:670)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:651)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:95)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        ... 27 more


BUILD FAILED in 48s
5 actionable tasks: 3 executed, 2 up-to-date
```

--- a/dev/cnf/build.bnd
+++ b/dev/cnf/build.bnd
@@ -100,29 +100,20 @@ instrument.taskInjection: false
     com.ibm.ws.componenttest:public.api\\;version=1.0.0,\
     com.ibm.ws.componenttest\\;version=latest}
 
-javac.bootclasspath.1.4: ${ibmjdklib.bootclasspath}
-javac.bootclasspath.1.6: ${ibmjdklib.bootclasspath}
-javac.bootclasspath.1.7: ${ibmjdklib.bootclasspath}
-javac.bootclasspath.1.8: ${ibmjdklib.bootclasspath}
+javac.bootclasspath.1.4: ${defaultjdklib.bootclasspath}
+javac.bootclasspath.1.6: ${defaultjdklib.bootclasspath}
+javac.bootclasspath.1.7: ${defaultjdklib.bootclasspath}
+javac.bootclasspath.1.8: ${defaultjdklib.bootclasspath}
 	
 javac.bootclasspath.java6:
 javac.bootclasspath.java7:
 javac.bootclasspath.java8:
 
-ibmjdklib.bootclasspath: \
-    ${java.home}/jre/lib/ibmcertpathfw.jar;version=file;boot=true, \
-    ${java.home}/jre/lib/ibmjcefw.jar;version=file;boot=true, \
-    ${java.home}/jre/lib/ibmjgssfw.jar;version=file;boot=true, \
-    ${java.home}/jre/lib/ibmjgssprovider.jar;version=file;boot=true, \
-    ${java.home}/jre/lib/ibmjssefw.jar;version=file;boot=true, \
-    ${java.home}/jre/lib/ibmorbapi.jar;version=file;boot=true, \
-    ${java.home}/jre/lib/ibmpkcs.jar;version=file;boot=true, \
-    ${java.home}/jre/lib/math.jar;version=file;boot=true, \
-    ${java.home}/jre/lib/rt.jar;version=file;boot=true, \
-    ${java.home}/jre/bin/default/jclSC180/vm.jar;version=file;boot=true, \
-    ${java.home}/jre/lib/xml.jar;version=file;boot=true, \
-    ${java.home}/jre/lib/xmldsigfw.jar;version=file;boot=true
-
+# catch-all if something goes wrong. This should be updated in
+# settings.gradle
+defaultjdklib.bootclasspath: \
+    ${java.home}/jre/lib/rt.jar;version=file;boot=true
+	
 internaljavac.bootclasspath.1.4: \
     com.ibm.ws.internal.prereq.java:java.rt;strategy=exact;version=1.6.0;boot=true
 

--- a/dev/settings.gradle
+++ b/dev/settings.gradle
@@ -76,11 +76,34 @@ def loadBuildProps() {
 
   def javaHome = org.gradle.internal.jvm.Jvm.current().getJavaHome()
   props.setProperty('java.home', javaHome.getAbsolutePath())
-  props.setProperty('javac.bootclasspath.1.4', '${ibmjdklib.bootclasspath}')
-  props.setProperty('javac.bootclasspath.1.6', '${ibmjdklib.bootclasspath}')
-  props.setProperty('javac.bootclasspath.1.7', '${ibmjdklib.bootclasspath}')
-  props.setProperty('javac.bootclasspath.1.8', '${ibmjdklib.bootclasspath}')
+
+  //scrape java.home JARs in jre/lib
+	
+  FileTree jdkLibs = fileTree(System.properties['java.home'])
+
+  //bin is needed for IBM JDKs
+  jdkLibs.include 'bin/default/**/vm.jar'
+  jdkLibs.include 'lib/*.jar'
+
+  def jdkLibString = ""
+  Boolean isFirstPass = true;
   
+  jdkLibs.each {File file -> 
+  
+
+  if (isFirstPass) {
+     jdkLibString = jdkLibString + file + ";version=file;boot=true"
+  } else {
+     jdkLibString = jdkLibString + ", " + file + ";version=file;boot=true"
+  }
+	 isFirstPass = false;
+  }
+
+  props.setProperty('javac.bootclasspath.1.4', jdkLibString)
+  props.setProperty('javac.bootclasspath.1.6', jdkLibString)
+  props.setProperty('javac.bootclasspath.1.7', jdkLibString)
+  props.setProperty('javac.bootclasspath.1.8', jdkLibString)
+ 
   if (usrHomeProps.exists() && !props.getProperty('artifactory.force.external.repo')) {
     //Some functionality is added if we are able to access particular internal systems.
     //This is to support more controlled dependency management, and multiple JDK paths for cross-compilation

